src/settingsView.ts

```diff
@@ -111,7 +115,15 @@ export class SettingsView extends DisposableBase implements vscodeTypes.WebviewV
   }
 
   private _updateSettings() {
-    void this._view!.webview.postMessage({ method: 'settings', params: { settings: this._settingsModel.json() } });
+    const settings = this._settingsModel.json();
+    const disabled = {
+      connectCopilot: !this._mcpConnection.isEnabled(),
+    };
+    const hidden = {
```

Comment:
Hidden UI is usually problematic. Let's instead disable and provide some information on hover, or upon an (i) button.